ggraph.Rmd

```diff
@@ -5,27 +5,159 @@ title: "ggplot2 extensions: ggraph"
 ### ggraph
 <https://github.com/thomasp85/ggraph>
 
-ggraph is an extension of ggplot2 tailored at plotting graph-like data structures (graphs, networks, trees, hierarchies...).
+ggraph extends the grammar of graphics provided by ggplot2 to cover graph and
+network data. This type of data consists of nodes and edges and are not 
+optimally stored in a single data.frame, as expected by ggplot2. Furthermore the
+spatial position of nodes (end thereby edges) are more often defined by the 
+graph structure through a layout function, rather than mapped to specific 
+parameters. ggraph handles all of these issues in an extensible way that lets
+the user gradually build up their graph visualization with different layers as
+expected from ggplot2, without needing to worry too much about how positions and
+other metrics are derived.
 
-```{r, message=FALSE,warning=FALSE, eval=FALSE}
-# Example from https://github.com/thomasp85/ggraph
-library(ggplot2)
-library(ggraph)
+While node-edge diagrams are the visualization people often relate to graph
+visualizations it is by far the only one, and ggraph have, or will have support
+for all types of common and uncommon visualization types.
+
+#### A simple example
 
-# igraph
-# We'll just make up some data
+```{r, eval=TRUE, echo=TRUE, warning=FALSE, message=FALSE}
+library(ggraph)
 library(igraph)
-gr <- erdos.renyi.game(n=20, p=0.2)
-E(gr)$weight <- sample(1:5, gsize(gr), TRUE)
-V(gr)$class <- sample(letters[1:3], gorder(gr), TRUE)
-
-# Plotting this is exactly as plotting a dendrogram
-# All igraph layouts are available - here we use Fructerman and Reingold
-ggraph(graph = gr, layout = 'fr') + 
-    geom_edge_link(aes(size = weight), color = 'grey', alpha = 0.5) + 
-    geom_node_point(aes(color = class), size = 10) + 
-    coord_fixed() + 
-    ggforce::theme_no_axes()
+
+# To show a simple example of the API we consider a hierarchy
+## Create a graph of the flare class hierarchy
+flareGraph <- graph_from_data_frame(flare$edges, vertices = flare$vertices)
+ggraph(flareGraph, 'igraph', algorithm = 'tree') + 
```

Comment:
I have just updated it and now I get this error:
"Error: ggplot2 doesn't know how to deal with data of class function"